ibis/expr/datatypes.py

```diff
@@ -1714,6 +1715,11 @@ def infer_integer(value: int, allow_overflow: bool = False) -> Integer:
     return int64
 
 
+@infer.register(enum.Enum)
+def infer_enum(value: enum.Enum) -> Enum:
```

Comment:
do we have a direct test of this anywhere?

Reply:
Err no...it is tested as part of the enum test I added but we should have a direct one. Adding now.

Reply:
Now tested in https://github.com/ibis-project/ibis/pull/2650/files#diff-752c3dd09545c70e427edd30421b0826bbeb836921c0ca5f6938756e90800ad8R440